dvc/external_repo.py

```diff
@@ -182,6 +182,7 @@ def pull_to(self, path, to_info):
 
     @contextmanager
     def open_by_relpath(self, path, mode="r", encoding=None, **kwargs):
+        """Opens a specified resource as a file descriptor"""
```

Comment:
Note: Added this docstring since external_repo seems like an important module. Kind of like the Repo class. They're very similar from what I can see.

Reply:
We don't return file descriptors (low-level concept), but rather file objects. I see that you got confused by `fd` in the code, it should really be `fobj`, of course.

Reply:
Right, sorry just a typo. `fd` is pretty a common var name for `with open()` so my bad. Updated!